File: backend/src/main/java/com/payroll/service/AttendanceService.java
package com.payroll.service;

import com.payroll.dto.response.AttendanceResponse;
import com.payroll.dto.response.MonthlyAttendanceSummaryResponse;
import com.payroll.dto.response.PageResponse;
import org.springframework.data.domain.Pageable;

public interface AttendanceService {
    AttendanceResponse checkIn(Long employeeId);
    AttendanceResponse checkOut(Long employeeId);
    PageResponse<AttendanceResponse> getHistory(Long employeeId, Pageable pageable);
    MonthlyAttendanceSummaryResponse getMonthlySummary(Long employeeId, int month, int year);
}

File: frontend/src/services/attendanceService.js
import api from './apiClient';

export const attendanceService = {
  checkIn: () => api.post('/attendance/check-in'),
  checkOut: () => api.post('/attendance/check-out'),
  history: (employeeId, page = 0, size = 10) =>
    api.get(`/attendance/history/${employeeId}?page=${page}&size=${size}`),
  monthlySummary: (employeeId, month, year) =>
    api.get(`/attendance/summary/${employeeId}?month=${month}&year=${year}`)
};
